src/gdcmodels/init_index.py

```diff
@@ -1,17 +1,42 @@
 #!/usr/bin/env python
 
 import argparse
+import dataclasses
+import logging
 import sys
-from typing import cast
+from typing import List, cast
 
 import elasticsearch
 from typing_extensions import Iterable, Optional, Protocol
 
 import gdcmodels
 
+logger = logging.getLogger("init_index")
+
+
+@dataclasses.dataclass
+class ESIndexBuilder:
```

Comment:
Thank you Jimmy for clarifying offline the reason why different logic is used for the non set indices aliasing as compared to the set indices aliasing. Specifically, you pointed out that there is a separate script that exists for creating the set indices which does not have access to the aliasing logic in the non-set indices script. This can be seen on steps 3c and 3d in the dev cluster sop: https://gdc-ctds.atlassian.net/wiki/spaces/GDC/pages/63897608/GDC+Cluster+SOP#:~:text=Now%20run%20the%20other%20script%20to%20create%20the%20saved%20sets%20indices%20(this%20should%20also%20be%20quick%20to%20run)%3A